airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -181,7 +181,8 @@ def _group_streams(
         state_manager = ConnectorStateManager(state=self._state)  # type: ignore  # state is always in the form of List[AirbyteStateMessage]. The ConnectorStateManager should use generics, but this can be done later
 
         name_to_stream_mapping = {
-            stream["name"]: stream for stream in self.resolved_manifest["streams"]
```

Comment:
The connector builder still calls `resolve_manifest` [here](https://github.com/airbytehq/airbyte-python-cdk/blob/a5a586020ee30c5c8a5ebadc06e03ef2c375a726/airbyte_cdk/connector_builder/connector_builder_handler.py#L98), which calls `source.resolved_manifest` [here](https://github.com/airbytehq/airbyte-python-cdk/blob/a5a586020ee30c5c8a5ebadc06e03ef2c375a726/airbyte_cdk/connector_builder/connector_builder_handler.py#L103).

I had done an experiment last hackdays to resolve in the frontend instead but it didn't end up working out, so this is still used.

Reply:
@maxi297 this is a very good callout. @lmossman then posted a conversation on how we could avoid having resolution logic in both FE and backend, and what needs to be done, and it would be worth following up on it.